### PR TITLE
chore: Proc macro to derive from_rpc and into_rpc

### PR DESCRIPTION
**Pull request by frondeus**
Fixes: #560

### CODE REVIEW

File: crates/utils/crates/rpc_dto/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+version = "0.1.0"
+edition = "2018"
+license = "Apache-2.0"
+
+[lib]
+proc-macro = true
+
+[dependencies]
+itertools = "0.10.0"
+proc-macro-crate = "1.0.0"
```
Comment (frondeus) [inline crates/utils/crates/rpc_dto/Cargo.toml:13]:
```suggestion
```

File: crates/utils/crates/rpc_dto/src/try_from_rpc.rs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                    })
+                }
+            }
+            Details::Transparent { field } => {
+                let name = field.ident.as_ref().unwrap();
+                quote! {
+                    Ok( Self { #name: TryFromRpc::try_from_rpc(rpc)? } )
+                }
+            }
+            Details::Struct(struct_) => struct_.synthesis(), // _ => todo!()
```
Comment (Kononnable) [inline crates/utils/crates/rpc_dto/src/try_from_rpc.rs:224]:
```suggestion
            Details::Struct(struct_) => struct_.synthesis(),
```

File: crates/utils/crates/rpc_dto/src/try_into_rpc.rs [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+                    })
+                }
+            }
+            Details::Transparent { field } => {
+                let name = field.ident.as_ref().unwrap();
+                quote! {
+                    TryIntoRpc::try_into_rpc(self.#name)
+                }
+            }
+            Details::Struct(struct_) => struct_.synthesis(), // _ => todo!()
```
Comment (Kononnable) [inline crates/utils/crates/rpc_dto/src/try_into_rpc.rs:234]:
```suggestion
            Details::Struct(struct_) => struct_.synthesis(),
```

File: crates/utils/crates/rpc_dto/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+authors = ["CDL Team"]
+version = "0.1.0"
+edition = "2018"
+license = "Apache-2.0"
+
+[lib]
+proc-macro = true
+
+[dependencies]
+itertools = "0.10.0"
```
Comment (Kononnable) [inline crates/utils/crates/rpc_dto/Cargo.toml:12]: Don't see it being used anywhere.

**Review by Kononnable — APPROVED** [review]

**Pull request merged**